label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -27,29 +30,15 @@
 
 public class LabelApplicationRuleConfigServiceImpl
     extends LabelApplicationRuleConfigServiceGrpc.LabelApplicationRuleConfigServiceImplBase {
-  static final String LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG =
-      "label.application.rule.config.service";
-  static final String MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT =
-      "max.dynamic.label.application.rules.per.tenant";
-  static final int DEFAULT_MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT = 100;
   private final IdentifiedObjectStore<LabelApplicationRule> labelApplicationRuleStore;
   private final LabelApplicationRuleValidator requestValidator;
-  private final int maxDynamicLabelApplicationRulesAllowed;
+  private final LabelApplicationRuleConfig labelApplicationRuleConfig;
 
   public LabelApplicationRuleConfigServiceImpl(
       Channel configChannel, Config config, ConfigChangeEventGenerator configChangeEventGenerator) {
-    int maxDynamicRules = DEFAULT_MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT;
-    if (config.hasPath(LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG)) {
-      Config labelApplicationRuleConfig =
-          config.getConfig(LABEL_APPLICATION_RULE_CONFIG_SERVICE_CONFIG);
-      if (labelApplicationRuleConfig.hasPath(MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT)) {
-        maxDynamicRules =
-            labelApplicationRuleConfig.getInt(MAX_DYNAMIC_LABEL_APPLICATION_RULES_PER_TENANT);
-      }
-    }
-    this.maxDynamicLabelApplicationRulesAllowed = maxDynamicRules;
+    this.labelApplicationRuleConfig = new LabelApplicationRuleConfig(config);
```

Comment:
Can you make the typed config object a constructor arg. If using that from the factory is a challenge, at least overload it so we can easily test this class with a mock config.

Reply:
Done